### PR TITLE
refactor: always set ics27 ports regardless of whether the capability exists

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

Need to update genesis unit tests

closes: #4626 

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/upgrades/genesis_test.go
Comment (colin-axner) [inline e2e/tests/upgrades/genesis_test.go:1]: copy/pasted from ics27 tests

File: modules/apps/27-interchain-accounts/controller/keeper/account.go [hunk 1]
```diff
@@ -62,7 +62,8 @@ func (k Keeper) registerInterchainAccount(ctx sdk.Context, connectionID, portID,
 	case k.portKeeper.IsBound(ctx, portID) && !k.hasCapability(ctx, portID):
 		return "", errorsmod.Wrapf(icatypes.ErrPortAlreadyBound, "another module has claimed capability for and bound port with portID: %s", portID)
 	case !k.portKeeper.IsBound(ctx, portID):
-		capability := k.BindPort(ctx, portID)
+		k.setPort(ctx, portID)
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/keeper/account.go:65]: ideally gets removed #4638 in the future

File: modules/apps/27-interchain-accounts/module.go [hunk 2]
```diff
@@ -123,19 +122,20 @@ func NewAppModule(controllerKeeper *controllerkeeper.Keeper, hostKeeper *hostkee
 // called once and as an alternative to InitGenesis.
 func (am AppModule) InitModule(ctx sdk.Context, controllerParams controllertypes.Params, hostParams hosttypes.Params) {
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/module.go:123]: #4639

**Review by damiannolan — APPROVED** [review]
Awesome, thank you @colin-axner! ❤️ 

Hopefully we can address these issues sooner rather than later :)

File: modules/apps/27-interchain-accounts/controller/keeper/genesis_test.go [hunk 3]
```diff
@@ -34,28 +53,47 @@ func (suite *KeeperTestSuite) TestInitGenesis() {
...
+
+			for _, port := range ports {
+				store := suite.chainA.GetContext().KVStore(suite.chainA.GetSimApp().GetKey(icacontrollertypes.StoreKey))
+				suite.Require().True(store.Has(icatypes.KeyPort(port)))
+
+				capability, found := suite.chainA.GetSimApp().ScopedICAControllerKeeper.GetCapability(suite.chainA.GetContext(), host.PortPath(port))
+				suite.Require().True(found)
+				suite.Require().NotNil(capability)
+			}
+		})
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/controller/keeper/genesis_test.go:94]: this is the main thing added

File: modules/apps/27-interchain-accounts/host/keeper/genesis_test.go [hunk 4]
```diff
@@ -4,15 +4,14 @@ import (
 	genesistypes "github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/genesis/types"
 	"github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/host/keeper"
 	"github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/host/types"
+	icahosttypes "github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/host/types"
 	icatypes "github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/types"
+	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
 	ibctesting "github.com/cosmos/ibc-go/v8/testing"
 )
 
 func (suite *KeeperTestSuite) TestInitGenesis() {
```
Comment (colin-axner) [inline modules/apps/27-interchain-accounts/host/keeper/genesis_test.go:13]: it's really hard to test both setups of capabilities already being initialized or not being initialized as the testing pkg will call init genesis first and the portID is hard coded. I personally think this sort of integration testing is better handled by the e2e's which is why I added coverage there

File: modules/apps/27-interchain-accounts/controller/keeper/genesis_test.go [hunk 5]
```diff
@@ -5,11 +5,29 @@ import (
 	"github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/controller/types"
 	genesistypes "github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/genesis/types"
 	icatypes "github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/types"
+	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
 	ibctesting "github.com/cosmos/ibc-go/v8/testing"
 )
 
 func (suite *KeeperTestSuite) TestInitGenesis() {
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/controller/keeper/genesis_test.go:12]: Table tests ++ 💪

File: e2e/tests/upgrades/genesis_test.go [hunk 6]
```diff
@@ -180,17 +180,6 @@ func (s *GenesisTestSuite) TestIBCGenesis() {
 
 			s.Require().NoError(test.WaitForBlocks(ctx, 10, chainA, chainB))
 		})
-
-		t.Run("verify tokens transferred", func(t *testing.T) {
```
Comment (colin-axner) [inline e2e/tests/upgrades/genesis_test.go]: this should be added with https://github.com/cosmos/ibc-go/issues/3476

**Review by chatton — APPROVED** [review]
Thanks for creating those clean-up issues, LGTM 👍

**Pull request merged**